site/learn/expert-advice/authentication/webauthn-explained.md

```diff
@@ -0,0 +1,515 @@
+
+```javascript
+options = {
+  publicKey: {
+    rpId: "cosmosclownstore.com",
+    challenge: "...",
+    userVerification: "preferred",
+    allowCredentials: ...
+    //...
+  }
+}
+```
+
+The options object has a few important properties:
+
+- the `publicKey` top-level key: this indicates we're using WebAuthn
+- the `rpId`: the website this request is for, such as cosmosclownstore.com
+- the `challenge`: a random string to prevent replay attacks
+- the `allowCredentials`: an array of credentials which are acceptable to this website
+
+There are other options as well, which allow the site to determine if the user should be fully verified and authenticated or just prove presence. That option will be discussed more below.
+
+After you call `get` with an `options` object, the response is returned as a promise. Here's an example response.
+
+```javascript
+{
+  authenticatorData: {
+    rpIdHash : ..., // hash of the requesting party Id
+    flags : ..., // bit array including authentication results
+    // other info
+  },
+  clientDataJSON: {
+    type: "webauthn.get",
+    challenge: ... // should match initial challenge
+  }
+}
+```
+
+If the response is successful, the response data will have an `authenticatorData` key, which has a `flags` member. `flags` is a bit array indicating if the user was present or verified, as well as extension data. The difference between presence and verification is covered below.
+
+You also receive a `clientDataJSON` object, which has the `challenge` and other information. In order to prevent replay attacks, verify that the `challenge` you receive is the same as the challenge you sent.
+
+## Authenticators, Revisited
+
+There are two types of authenticators: cross-platform authenticators and platform authenticators. 
+
+Cross-platform authenticators are hardware devices that can be used between different devices, like a YubiKey.
+
+Platform authenticators are built into the operating system and exist in all major operating systems, both desktop and mobile. They typically have a secure hardware enclave where private keys are stored. The ubiquity of platform authenticators means they are best for consumer-facing applications. The operating systems have sophisticated biometric authentication; you can access these with WebAuthn using the standard JavaScript API outlined above.
+
+Authenticators can also be implemented entirely in software, such as [this one](https://chrome.google.com/webstore/detail/virtual-authenticators-ta/gafbpmlmeiikmhkhiapjlfjgdioafmja), but these are typically used for testing, not a real-world authentication scenario.
+
+WebAuthn handles communication between the browser and the website, and the browsers leverage the CTAP2 protocol to talk to authenticators. As a WebAuthn system implementer, you don't need to know CTAP2, just understand it exists.
+
+As mentioned above, there are three main WebAuthn ceremonies: authentication, which you've already seen, registration and attestation.
```

Comment:
Sorry if you're still in the process of updating this, but as far as I can tell, attestation is not a separate ceremony.

As the Relying Party, you wouldn't say "please provide me attestation" in a vacuum. The RP says, "okay, you want to register a new credential. Please provide attestation along with the credential data" as part of the registration ceremony.